Advanced-Building-Blocks---Enumerable

BrianSammit opened the pull request:
We duplicated #each, #each_with_index, #select, #all, #any, #none, #count, #map and #inject.

We disable some errors from rubocop that doesn't affect the program.

elasfarc committed:
setup linters

elasfarc committed:
my_each_test

logic for my_each that till now works only with array

BrianSammit committed:
Add a method my_each_with_index

elasfarc committed:
[test_hash.rb] my_each and my_each_index could accept a hash and a range

BrianSammit committed:
Add my_select method

elasfarc committed:
my_all? limted-> without arguments and a block must be given

elasfarc committed:
my_all? limted no arguments

handling no block given by calling itself inside

elasfarc committed:
my_all? completed (3 pathes ) [argument, no-block-given, normal case]

elasfarc committed:
change files names so they don't contain '?'

BrianSammit committed:
Add my_any method

BrianSammit committed:
Add my_any to main page

BrianSammit committed:
Add my_none method

elasfarc committed:
Create my_count.rb

elasfarc committed:
Update my_none.rb

   #2nd step dealing with the case when no argument and no block

BrianSammit committed:
fixed problem

BrianSammit committed:
Add my_map method

BrianSammit committed:
Add my_map method

elasfarc committed:
add my_inject

elasfarc committed:
my_map [Proc & Block ]

BrianSammit committed:
tested the methods

BrianSammit committed:
Update README.md

elasfarc committed:
fix lint errors

elasfarc committed:
Merge branch 'feature-branch' of https://github.com/elasfarc/Advanced-Building-Blocks---Enumerables into feature-branch

elasfarc committed:
Delete my_all.rb

andresporras3423 reviewed (changes requested):
### Status: Changes Required :recycle:
**Hello Mo Al-Asfar & Brian Cruz Rodriguez** 
Nice job, your project passes most of the automatic tests :muscle:  :muscle: 
We still need to fix some details:

- [x] enumerables #my_select returns an enumerator if no block is given
`[1,2,3,4].my_select.class==Enumerator`
- [x] enumerables #my_all? when no block or argument is given returns true when none of the collection members are false or nil
`[true, [false]].all? == [true, [false]].my_all?`
- [x] enumerables #my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Integer
`[1,2,3].all?(Integer) == [1,2,3].my_all?(Integer)`
- [x] enumerables #my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric
`[1,-2,3.4].all?(Numeric) == [1,-2,3.4].my_all?(Numeric)`
- [x] enumerables #my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
`['dog', 'cat'].all?(/d/) == ['dog', 'cat'].my_all?(/d/)`
- [x] enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true
`[false, 0].any? == [false, 0].my_any?`
- [x] enumerables #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric
`[1,'',[]].any?(Numeric) == [1,'',[]].my_any?(Numeric)`
- [x] enumerables #my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex
`['dog', 'cat'].none?(/d/) == ['dog', 'cat'].my_none?(/d/)`
- [x]  enumerables #my_any? when a Regex is passed as an argument returns true if any of the collection matches the Regex
`['dog', 'cat'].any?(/d/) == ['dog', 'cat'].my_any?(/d/)`
- [x] # enumerables #my_inject raises a "LocalJumpError" when no block or argument is given
```
begin
   [1,2,3,4].my_inject
rescue LocalJumpError
   true
end
```
- [x] enumerables #my_inject when a block is given with an initail value combines all elements of enum by applying a binary operation, specified by a block
`(1..3).inject(4) { |prod, n| prod * n } == (1..3).my_inject(4) { |prod, n| prod * n }`
- [x] enumerables #my_inject when a symbol is specified with an initail value combines each element of the collection by applying the symbol as a named method::array
`[1,2,3].inject(4, :*) == [1,2,3].my_inject(4, :*)`
- [x]  enumerables #my_inject when a symbol is specified with an initail value combines each element of the collection by applying the symbol as a named method::range
`(1..3).inject(4, :*) == (1..3).my_inject(4, :*)`

Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

BrianSammit committed:
fixed my_select method

BrianSammit committed:
removed merge conflicts

BrianSammit committed:
fixed my_select method

elasfarc committed:
update enmerables

my_all? when no block or argument is given returns true when none of the collection members are false or nil
 #my_none?, my_all? & my_any? when a Regex is passed as an argument returns false if any of the collection matches the Regex
#my_inject when a block is given with an initail value combines all elements of enum by applying a binary operation, specified by a block
#my_inject when a symbol is specified with an initail value combines each element of the collection by applying the symbol as a named method::array

elasfarc committed:
lint errors

BrianSammit committed:
resolve errors

BrianSammit committed:
fixed erros with merge

andresporras3423 reviewed (changes requested):
### Status: Changes Required :recycle:
**Hello Mo Al-Asfar & Brian Cruz Rodriguez** 
Nice job with the improvements to your project :muscle:  :muscle: 
We still need to fix some issues:

- [x] enumerables #my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
`[5,5,5].all?(5) == [5,5,5].my_all?(5)`
- [x] enumerables #my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern
`[5,5,[5]].all?(5) == [5,5,[5]].my_all?(5)`
- [x] enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true
`[false, 0].any? == [false, 0].my_any?`
- [x] enumerables #my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
`['dog','car'].any?('cat') == ['dog','car'].my_any?('cat')`
- [x] enumerables #my_any? when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern
`['cat', 'dog','car'].any?('cat') == ['cat', 'dog','car'].my_any?('cat')`
- [x] enumerables #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
`['dog','car'].none?(5) == ['dog','car'].my_none?(5)`
- [x] enumerables #my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern
`[5,'dog','car'].none?(5) == [5,'dog','car'].my_none?(5)`
- [ ] enumerables #my_inject raises a "LocalJumpError" when no block or argument is given
```
begin
  [1,2,3,4].my_inject
rescue LocalJumpError
  true
end
```

Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

BrianSammit committed:
Fixed logic problem in my_all, my_any, my_none and my_inject

BrianSammit committed:
fixed linters errors

BrianSammit committed:
fixed linters

kubilaycaglayan reviewed (approved):
Hello Team :wave: 
Status: Project Approved :green_circle: 

Great job on making the changes. :+1: 
Your methods are working amazingly! :+1: :blush: 

With all my respect... :pray:
